Tools/AP_Periph/can.cpp

```diff
@@ -977,79 +994,112 @@ static bool shouldAcceptTransfer(const CanardInstance* ins,
     return false;
 }
 
+static void cleanup_stale_trx(uint64_t &timestamp_usec)
+{
+    for (uint8_t i = 0; i < HAL_NUM_CAN_IFACES; i++) {
+        canardCleanupStaleTransfers(&canard[i], timestamp_usec);
+    }
+}
+
+static int16_t canard_broadcast(uint64_t data_type_signature,
+                                uint16_t data_type_id,
+                                uint8_t priority,
+                                const void* payload,
+                                uint16_t payload_len)
+{
+    int16_t cache = -1;
+    bool success = false;
+    for (uint8_t i = 0; i < HAL_NUM_CAN_IFACES; i++) {
+        if (canardGetLocalNodeID(&canard[i]) == CANARD_BROADCAST_NODE_ID) {
+            continue;
+        }
+        cache = canardBroadcast(&canard[i],
+                        data_type_signature,
+                        data_type_id,
+                        &transfer_id[i],
+                        priority,
+                        payload,
+                        payload_len);
+        if (cache == 0) {
+            // we had atleast one interface where transaction was successful
+            success = true;
+        }
+    }
+    return success?0:cache;
```

Comment:
given two IFACES.. first one is successful, second one fails. The returned result will be the failed one

Reply:
No, it should be successful, the second one is most likely to fail if it is disconnected.

Reply:
So a disconnected iface 2 cache value is what you want to return?

Reply:
we return cache on fail, (success == false). if (success == true) its 0, which means successful. If we were to fail we would return the last fail value cached.

Reply:
Ahh yes. Sorry. I misread it.